ssl-config/src/main/java/com/palantir/remoting1/config/ssl/PemX509Certificate.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting1.config.ssl;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE, jdkOnly = true)
+@JsonSerialize(as = ImmutablePemX509Certificate.class)
+@JsonDeserialize(as = ImmutablePemX509Certificate.class)
+public abstract class PemX509Certificate {
+
+    /** A X509.1 certificate in PEM format encoded in UTF-8. */
+    public abstract String pemCertificate();
```

Comment:
they can be anything - `PUBLIC KEY`, `CERTIFICATE`, etc.. I don't think it actually matters. Can write test if you wish?